File: tests/machine/x/clj/list_nested_assign.clj
(ns main)

(defn _indexList [xs i]
  (let [idx (if (neg? i) (+ i (count xs)) i)]
    (if (or (< idx 0) (>= idx (count xs)))
      (throw (ex-info "index out of range" {}))
      (nth xs idx))))

(declare matrix)

(defn -main []
  (def matrix [[1 2] [3 4]]) ;; list of list of int
  (def matrix (assoc-in matrix [1 0] 5)) ;; int
  (println (_indexList (_indexList matrix 1) 0))
)

(-main)

File: tests/machine/x/cs/list_nested_assign.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        List<List<int>> matrix = new List<List<int>> { new List<int> { 1, 2 }, new List<int> { 3, 4 } };
        matrix[1][0] = 5;
        Console.WriteLine(_indexList(_indexList(matrix, 1), 0));
    }
    static dynamic _indexList(dynamic l, long i) {
        var list = l as System.Collections.IList;
        if (list == null) throw new Exception("index() expects list");
        if (i < 0) i += list.Count;
        if (i < 0 || i >= list.Count) throw new Exception("index out of range");
        return list[(int)i];
    }
    
}
